src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTest.cs

```diff
@@ -3,580 +3,381 @@
-            p.WaitForExit();
-            Assert.False(s_Process_EnableRaiseEvents_isExitedEventHandlerCalled, String.Format("Process_CanRaiseEvents0002: {0}", "isExited Event called with the default settings for EnableRaiseEvents"));
+                Process p = CreateProcessInfinite();
+                p.EnableRaisingEvents = true;
+                p.Exited += p_Exited;
+                StartAndKillProcessWithDelay(p);
+                Assert.True(s_Process_EnableRaiseEvents_isExitedEventHandlerCalled, String.Format("Process_CanRaiseEvents0001: {0}", "isExited Event not called when EnableRaisingEvent is set to true."));
+            }
 
-            s_Process_EnableRaiseEvents_isExitedEventHandlerCalled = false;
+            {
+                s_Process_EnableRaiseEvents_isExitedEventHandlerCalled = false;
 
-            // Same test, this time explicitly set the property to false
-            p = CreateProcessInfinite();
-            p.EnableRaisingEvents = false;
-            p.Exited += p_Exited;
-            p.Start();
-            p.Kill();
-            p.WaitForExit();
-            Assert.False(s_Process_EnableRaiseEvents_isExitedEventHandlerCalled, String.Format("Process_CanRaiseEvents0003: {0}", "isExited Event called with the EnableRaiseEvents = false"));
+                // Check with the default settings (false, events will not be raised)
+                Process p = CreateProcessInfinite();
+                p.Exited += p_Exited;
+                StartAndKillProcessWithDelay(p);
+                Assert.False(s_Process_EnableRaiseEvents_isExitedEventHandlerCalled, String.Format("Process_CanRaiseEvents0002: {0}", "isExited Event called with the default settings for EnableRaiseEvents"));
+            }
+
+            {
+                s_Process_EnableRaiseEvents_isExitedEventHandlerCalled = false;
 
-            s_Process_EnableRaiseEvents_isExitedEventHandlerCalled = false;
+                // Same test, this time explicitly set the property to false
+                Process p = CreateProcessInfinite();
+                p.EnableRaisingEvents = false;
+                p.Exited += p_Exited;
+                StartAndKillProcessWithDelay(p);
+                Assert.False(s_Process_EnableRaiseEvents_isExitedEventHandlerCalled, String.Format("Process_CanRaiseEvents0003: {0}", "isExited Event called with the EnableRaiseEvents = false"));
+            }
         }
 
-        static void p_Exited(object sender, EventArgs e)
+        void p_Exited(object sender, EventArgs e)
         {
             s_Process_EnableRaiseEvents_isExitedEventHandlerCalled = true;
         }
 
         [Fact]
-        public static void Process_ExitCode()
+        public void Process_ExitCode()
         {
-            Process p = CreateProcess();
-            p.Start();
-            p.WaitForExit();
-            Assert.Equal(p.ExitCode, 100);
+            {
+                Process p = CreateProcess();
+                p.Start();
+                DelayTask();
```

Comment:
Why are we not waiting for exit here?  Seems like we're relying on timing to ensure that the process has exited.